frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -872,14 +873,23 @@ export async function testAuth() {
 export async function setCluster(clusterReq: ClusterRequest) {
   return request(
     '/cluster',
-    { method: 'POST', body: JSON.stringify(clusterReq), headers: JSON_HEADERS },
+    {
+      method: 'POST',
+      body: JSON.stringify(clusterReq),
+      headers: { ...JSON_HEADERS, ...getHeadlampAPIHeaders() },
```

Comment:
Looks good! I wonder if it'd be easier to have a request variant that already puts these headers in `headlampApiRequest`.
Not super convinced this is much more useful, so take it as a light suggestion and feel free to decide.